### PR TITLE
Set CPU quota on agent

### PR DESCRIPTION
**Pull request by narrieta**

### CODE REVIEW

File: azurelinuxagent/common/cgroupconfigurator.py [hunk 1]
```diff
@@ -309,7 +322,7 @@ def __ensure_azure_slices_exist(self):
                     for unit_file in files_to_create:
                         self._cleanup_unit_file(unit_file)
 
-            return _AZURE_SLICE if return_azure_slice else "system.slice"
+            logger.info("Agent's CPUQuota: {0}",   self._cgroups_api.get_unit_property(agent_unit_name, "CPUQuotaPerSecUSec"))
```
Comment (narrieta) [inline azurelinuxagent/common/cgroupconfigurator.py:325]: When setting, the property is named CPUQuota (see line 69); when retrieving it is named CPUQuotaPerSecUSec

File: tests/common/mock_cgroup_commands.py [hunk 2]
```diff
@@ -151,14 +171,33 @@ def mock_open(path, *args, **kwargs):
...
+        (os.path.join(data_dir, 'init', 'walinuxagent.service'), UnitFilePaths.walinuxagent),
+        (os.path.join(data_dir, 'init', 'azure.slice'), UnitFilePaths.azure),
+        (os.path.join(data_dir, 'init', 'azure-vmextensions.slice'), UnitFilePaths.vmextensions)
+    ]
+
+    def add_data_file(source, target):
+        shutil.copyfile(source, get_mapped_path(target))
+
     builtin_popen = "__builtin__.open" if sys.version_info[0] == 2 else "builtins.open"
-    with patch("azurelinuxagent.common.cgroupapi.subprocess.Popen", side_effect=mock_popen) as patcher:
```
Comment (narrieta) [inline tests/common/mock_cgroup_commands.py:155]: For some reason i can't remember i was returning the Popen patcher but there are no more uses of it in the tests. Now the return value is the function itself, which is used to attach the properties on lines 191-198 in the new code

File: azurelinuxagent/common/cgroupconfigurator.py (hunk 1 above)
Comment (larohra) [reply azurelinuxagent/common/cgroupconfigurator.py:325]: Is this a Cgroups thing?

Comment (narrieta) [reply azurelinuxagent/common/cgroupconfigurator.py:325]: yup; i added a comment in the next iteration of the code; i'll submit the PR later this week

File: tests/common/mock_cgroup_commands.py [hunk 3]
```diff
@@ -67,23 +71,28 @@
...
     r"^(/lib/systemd/system)",
     r"^(/etc/systemd/system)"
 ]
 
-# W0212: Access to a protected member __commands of a client class (protected-access) - Disabled: patcher.__commands is added
-# only for debugging purposes and should not be public (hence it is marked as private).
-# pylint: disable=protected-access
+
+#  Class has no __init__ method (no-init): Disabled; the class is just an enum
+class UnitFilePaths:  # pylint: disable=no-init
```
Comment (larohra) [inline tests/common/mock_cgroup_commands.py:87]: In favor with removing this pylint too globally?

File: tests/common/mock_cgroup_commands.py [hunk 4]
```diff
@@ -151,14 +173,34 @@ def mock_open(path, *args, **kwargs):
...
         with patch("azurelinuxagent.common.cgroupconfigurator.fileutil.mkdir", side_effect=mock_mkdir):
             with patch("azurelinuxagent.common.cgroupapi.os.path.exists", side_effect=mock_path_exists):
                 with patch(builtin_popen, side_effect=mock_open):
                     with patch('azurelinuxagent.common.cgroupapi.CGroupsApi.cgroups_supported', return_value=True):
                         with patch('azurelinuxagent.common.cgroupapi.CGroupsApi.is_systemd', return_value=True):
-                            patcher.__commands = __MOCKED_COMMANDS[:]
-                            patcher.add_command_mock = add_command_mock
-                            patcher.get_mapped_path = get_mapped_path
-                            yield patcher
+                            class Mock:
```
Comment (larohra) [inline tests/common/mock_cgroup_commands.py:196]: Rename to `CgroupMock` maybe? Just to make it more clear

**Review by larohra — APPROVED** [review]
small cosmetic comments, else LGTM

File: tests/common/mock_cgroup_commands.py (hunk 3 above)
Comment (narrieta) [reply tests/common/mock_cgroup_commands.py:87]: sounds good; will submit a PR

File: tests/common/mock_cgroup_commands.py (hunk 4 above)
Comment (narrieta) [reply tests/common/mock_cgroup_commands.py:196]: It's gone in the next iteration of the mocks; will submit the PR shortly

**Pull request merged**